### ISSUE TITLE
undefined symbol: sv_dup_inc

### ISSUE THREAD

**Issue by eserte** [Leont/crypt-bear#2]
On some of my smokers the test suite fails like this:
```
#   Failed test 'Crypt/Bear.pm loaded ok'
#   at t/00-compile.t line 77.
#          got: '65280'
#     expected: '0'
Can't load '/home/cpansand/.cpan/build/2024111220/Crypt-Bear-0.001-0/blib/arch/auto/Crypt/Bear/Bear.so' for module Crypt::Bear: /home/cpansand/.cpan/build/2024111220/Crypt-Bear-0.001-0/blib/arch/auto/Crypt/Bear/Bear.so: undefined symbol: sv_dup_inc at /opt/perl-5.38.2D/lib/5.38.2/aarch64-linux-ld/DynaLoader.pm line 206.
Compilation failed in require at -e line 1.
... etc ...
```

**Comment by eserte** [comment]
On other (seen on some freebsd setups) the compilation already fails:
```
cc -c  -Iinclude -Isrc  -DHAS_FPSETMASK -DHAS_FLOATINGPOINT_H -DNO_POSIX_2008_LOCALE -fno-strict-aliasing -pipe -fstack-protector-strong -I/usr/local/include -O2 -pipe    -DVERSION=\"0.001\"  -DXS_VERSION=\"0.001\" -DPIC -fPIC "-I/usr/perl5.40.0/lib/5.40.0/aarch64-freebsd/CORE"   lib/Crypt/Bear.c -o lib/Crypt/Bear.o
lib/Crypt/Bear.xs:484:19: warning: call to undeclared function 'sv_dup_inc'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
  484 |         self->callback = sv_dup_inc(decoder->callback, params);
      |                          ^
lib/Crypt/Bear.xs:484:17: error: incompatible integer to pointer conversion assigning to 'SV *' (aka 'struct sv *') from 'int' [-Wint-conversion]
  484 |         self->callback = sv_dup_inc(decoder->callback, params);
      |                        ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
lib/Crypt/Bear.xs:486:16: error: incompatible integer to pointer conversion assigning to 'SV *' (aka 'struct sv *') from 'int' [-Wint-conversion]
  486 |                 self->buffer = sv_dup_inc(decoder->buffer, params);
      |                              ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
lib/Crypt/Bear.xs:488:14: error: incompatible integer to pointer conversion assigning to 'SV *' (aka 'struct sv *') from 'int' [-Wint-conversion]
  488 |                 self->name = sv_dup_inc(decoder->name, params);
      |                            ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
...
```

**Comment by eserte** [comment]
Actually it seems to pass only with threaded perls.

**Comment by Leont** [comment]
Fixed in 0.002
